mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -504,8 +504,8 @@ trait Completions { this: MetalsGlobal =>
         )
       case (m @ Match(_, Nil)) :: parent :: _ =>
         CaseKeywordCompletion(m.selector, editRange, pos, text, parent)
-      case Ident(name) :: (_: CaseDef) :: (m: Match) :: parent :: _
-          if isCasePrefix(name) =>
+      case Ident(name) :: (cd: CaseDef) :: (m: Match) :: parent :: _
```

Comment:
So, the problem is we have the same path for 

```
Option(1) match {
  case Some(a) => cas@@
``` and

```
Optoin(1) match {
  case Some(a) => a
  cas@@
```

am I right?
Is it possible to check the `CaseDef`'s right hand side and the `Ident(name)` is the same symbol or not?

Reply:
Even if the indentation is incorrect, like here
```scala 
Option(1) match {
  case Some(a) => 
cas@@ 
```
This `cas` is still a part of `CaseDef` RHS. We could do a condition to not suggest this completion if `CaseDef` RHS == next `Ident`, but this would not allow sth like
```scala 
Option(1) match {
  case Some(a) => 
  cas@@ 
```
and some people would want to first list cases, then handle them

Reply:
> some people would want to first list cases, then handle them

Definitely, that how I usually do this

Reply:
>> some people would want to first list cases, then handle them
> Definitely, that how I usually do this


Interesting, I didn't assume people write case clauses like that 👍 
(actually, I was thinking we don't need to offer case completions in the following scenario).

```
Option(1) match {
  case Some(a) => 
cas@@ 
```

However, if people wanna completion like that way, my suggestion doesn't work.

Looks like current implementation is the best workaround :+1:

Reply:
> Interesting, I didn't assume people write case clauses like that 👍
(actually, I was thinking we don't need to offer case completions in the following scenario).

It depends on what you're trying to do. If you need to handle Option then you'll probably go with exhaustive option, but sometimes, when you're matching on sealed hierarchy with multiple variants you might want to handle only one or two of them, and treat rest as the `_`. This is just an example, but having completions in situations like above is really nice. Such details make huge difference in UX, at least for me. 
That's being said, thank you @jkciesluk for yur work on this 🚀